src/search.cpp

```diff
@@ -1491,7 +1500,7 @@ Value Search::Worker::search(
                        bestValue >= beta    ? BOUND_LOWER
                        : PvNode && bestMove ? BOUND_EXACT
                                             : BOUND_UPPER,
-                       depth, bestMove, unadjustedStaticEval, tt.generation());
+                       moveCount != 0 ? depth : std::min(MAX_PLY - 1, depth + 6), bestMove, unadjustedStaticEval, tt.generation());
```

Comment:
Curious why we add 6 in particular?  Is it the smallest constant that fixes the problem?

Reply:
I copied this code/constant from vondele 's Test. IMO the smallest constant that should fix the problem is 4. (We also could apply it only to stalemates instead to applying it to mates too)

Reply:
I though to treat it like we treat a table base probe, i.e. both are definitive results, so it the same as we do there already in the code.

Reply:
https://github.com/official-stockfish/Stockfish/blob/d7c04a942950f1fe3f655bf8b608e8ef21c07628/src/search.cpp#L770-L772